src/java/htsjdk/samtools/SAMFileWriterFactory.java

```diff
@@ -311,28 +332,91 @@ public CRAMFileWriter makeCRAMWriter(final SAMFileHeader header, final OutputStr
         return writer;
     }
 
+    /**
+     * Create a CRAMFileWriter on an output file. Requires input record to be presorted to match the
+     * sort order defined by the input header.
+     *
+     * Note: does not honor factory settings for USE_ASYNC_IO.
+     * Read names and tags are preserved.
+     *
+     * @param header entire header. Sort order is determined by the sortOrder property of this arg.
+     * @param outputFile where to write the output.  Must end with .sam, .bam or .cram.
+     * @param referenceFasta reference sequence file
+     * @return CRAMFileWriter
+     *
+     */
     public CRAMFileWriter makeCRAMWriter(final SAMFileHeader header, final File outputFile, final File referenceFasta) {
+        return createCRAMWriter(header, true, outputFile, referenceFasta);
+    }
 
-        final boolean createIndex = this.createIndex && IOUtil.isRegularPath(outputFile);
-        if (this.createIndex && !createIndex) {
-            System.err.println("Cannot create index for CAM because output file is not a regular file: " + outputFile.getAbsolutePath());
-        }
 
-        try {
+    /**
+     * Create a CRAMFileWriter on an output file.
+     *
+     * Note: does not honor factory setting for USE_ASYNC_IO.
+     * Read names and tags are preserved.
+     *
+     * @param header entire header. Sort order is determined by the sortOrder property of this arg.
+     * @param presorted  if true, SAMRecords must be added to the SAMFileWriter in order that agrees with header.sortOrder.
+     * @param outputFile where to write the output.  Must end with .sam, .bam or .cram.
+     * @param referenceFasta reference sequence file
+     * @return CRAMFileWriter
+     *
+     */
+    public CRAMFileWriter makeCRAMWriter(final SAMFileHeader header, final boolean presorted, final File outputFile, final File referenceFasta) {
+        return createCRAMWriter(header, presorted, outputFile, referenceFasta);
+    }
+
+    /**
+     * Create a CRAMFileWriter on an output file.
+     *
+     * Note: does not honor factory setting for USE_ASYNC_IO.
+     *
+     * @param header entire header. Sort order is determined by the sortOrder property of this arg.
+     * @param presorted  if true, SAMRecords must be added to the SAMFileWriter in order that agrees with header.sortOrder.
+     * @param outputFile where to write the output.  Must end with .sam, .bam or .cram.
+     * @param referenceFasta reference sequence file
+     * @return CRAMFileWriter
+     *
+     * Read names and tags are preserved.
+     */
+    private CRAMFileWriter createCRAMWriter(final SAMFileHeader header, final boolean presorted, final File outputFile, final File referenceFasta) {
```

Comment:
There are 2 overloads of makeCramWriter that call into this method, and a different one that doesn't but duplicates some of the work that this one does.  Could you unify them?  It's seems very likely that they will eventually become confusingly out of sink if they don't call into the same code.